WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/DiscountLineDetailsView.swift

```diff
@@ -0,0 +1,136 @@
+import SwiftUI
+
+struct DiscountLineDetailsView: View {
+
+    @ObservedObject private var viewModel: FeeOrDiscountLineDetailsViewModel
+
+    init(viewModel: FeeOrDiscountLineDetailsViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: .zero) {
+            Text(Localization.fixedPriceDiscountLabel)
+                .renderedIf(viewModel.feeOrDiscountType == .fixed)
+                .padding()
+            Text(Localization.percentagePriceDiscountLabel)
+                .renderedIf(viewModel.feeOrDiscountType == .percentage)
+                .padding()
+            HStack {
+                inputFixedField
+                    .renderedIf(viewModel.feeOrDiscountType == .fixed)
+                inputPercentageField
+                    .renderedIf(viewModel.feeOrDiscountType == .percentage)
+                HStack(spacing: 0) {
+                    Button(viewModel.currencySymbol) {
+                        viewModel.feeOrDiscountType = .fixed
+                    }
+                    .buttonStyle(isPrimary: viewModel.feeOrDiscountType == .fixed)
+                    .fixedSize(horizontal: true, vertical: false)
+                    Button(viewModel.percentSymbol) {
+                        viewModel.feeOrDiscountType = .percentage
+                    }
+                    .buttonStyle(isPrimary: viewModel.feeOrDiscountType == .percentage)
+                    .fixedSize(horizontal: true, vertical: false)
+                }
+            }
+        }
+    }
+
+    private var inputFixedField: some View {
+        AdaptiveStack(horizontalAlignment: .leading) {
+            HStack {
+                BindableTextfield(Localization.fixedPriceDiscountInputPlaceholder,
+                                  text: $viewModel.amount,
+                                  focus: .constant(true))
+                .keyboardType(.numbersAndPunctuation)
+            }
+        }
+        .frame(maxWidth: .infinity, minHeight: Layout.rowHeight)
+        .padding([.leading, .trailing], Layout.padding)
+        .overlay {
+            RoundedRectangle(cornerRadius: Layout.frameCornerRadius)
+                .inset(by: 0.25)
```

Comment:
perhaps this should be in the Layout enum?

Reply:
Yes! Fixed: ea5b5b3caad410eda9e06624f5b77458d44dfdd5